internal/venice-client-common/src/main/java/com/linkedin/venice/compute/ComputeUtils.java

```diff
@@ -41,11 +48,22 @@ public class ComputeUtils {
   private static final RedundantExceptionFilter REDUNDANT_EXCEPTION_FILTER =
       RedundantExceptionFilter.getRedundantExceptionFilter();
 
-  public static void checkResultSchema(
-      Schema resultSchema,
-      Schema valueSchema,
-      int version,
-      List<ComputeOperation> operations) {
+  /**
+   * N.B.: This deserializer performs an evolution from the schema of {@link ComputeRequestV3} to that of
+   * {@link ComputeRequest}, with the only difference between the two being that the items of the operations list
+   * in the former are defined as a union of one type, which unfortunately results in the SpecificRecord typing this
+   * as a {@link List<Object>}. This is a design shortcoming, but which we cannot easily fix, since there are already
+   * clients using this protocol. On the server-side, however, we wish to use proper types without lots of casting,
+   * which we can achieve by letting Avro do the evolution.
+   */
+  private static final RecordDeserializer<ComputeRequest> DESERIALIZER =
+      getFastAvroSpecificDeserializer(ComputeRequestV3.SCHEMA$, ComputeRequest.class);
+
+  public static ComputeRequest deserializeComputeRequest(BinaryDecoder decoder, ComputeRequest reuse) {
```

Comment:
Same question here.
If v5 adds a new optional field in the top-level, we have to use the right write schema to decode it, right?

Reply:
Yes that's correct.